### PR TITLE
Restructuring/Slimming down carousel functionality page

### PR DESCRIPTION
**Pull request by yatil**

### CODE REVIEW

File: source/carousels/functionality.html.erb.md [hunk 1]
```diff
@@ -9,19 +9,14 @@ wcag_success_criteria:
   - 4.1.2
 ---
 
-Provide functionality using scripting to display the carousel items one at a time, and to allow users to browse through them.
+Carusels display the one content item at a time while hiding the others. They allow users to browse through them.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: This should probably be “one of multiple content items”.

File: source/carousels/functionality.html.erb.md [hunk 2]
```diff
@@ -9,19 +9,14 @@ wcag_success_criteria:
...
-## Displaying carousel items
+## Position slides
 
-Carousel items that are visually hidden should also be hidden from assistive technology, to avoid a mismatch between what is visually on the screen and what the user is interacting with. All carousel items that are not displayed visually are hidden by using the CSS `display: none` declaration.
-
-### Styling when scripting is enabled
-{:.ex}
-
-In the example below, JavaScript is used to add the class name `active` to the carousel container. The styling hides all the slides in a stack with only the slide with the class name `current` being visible on top. That class name is added to the item using JavaScript as well.
+If the JavaScript is enabled, a class `active` is added to the carousel region so we can distinguish between that state and the non-JavaScript fallback. The slides are positioned absolutely in the carousel, and the current slide is positioned frontmost using `z-index`.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: * “region so we can apply CSS in the case JavaScript is activated.”
* “positioned using `position: absolute`”

File: source/carousels/functionality.html.erb.md [hunk 3]
```diff
@@ -159,21 +154,9 @@ The outcome looks like this:
 
 ## Adding previous and next buttons
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: Add previous and next buttons (to be more consistent)

File: source/carousels/functionality.html.erb.md [hunk 4]
```diff
@@ -159,21 +154,9 @@ The outcome looks like this:
...
-
-As with the text, a semi-transparent white background is used to ensure sufficient color contrast between the black arrows and the slide images. The buttons also increase in size when users hover over or focuses them. This provides more click area for people with reduced dexterity and better highlight where the current focus is.
-
-{::nomarkdown}
-<%= notes_start %>
-{:/nomarkdown}
-
-**Note:** If the carousel uses `<a>` instead of `<button>` elements, a `href` attribute needs to be present to enable keyboard access. Additionally `role="button"` should be applied to the link so assistive technology users know that they expect an interaction on the page rather than a link to another page.
+Add `<button>` elements that allow users to switch back and forth between carousel items. They can be individually styled and also provide semantic meaning. Such elements are also compatible with assistive technology and keyboard use.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/such elements/These buttons
s/also compatible/also work
s/use/use out of the box

File: source/carousels/functionality.html.erb.md [hunk 5]
```diff
@@ -159,21 +154,9 @@ The outcome looks like this:
...
-<%= notes_start %>
-{:/nomarkdown}
-
-**Note:** If the carousel uses `<a>` instead of `<button>` elements, a `href` attribute needs to be present to enable keyboard access. Additionally `role="button"` should be applied to the link so assistive technology users know that they expect an interaction on the page rather than a link to another page.
+Add `<button>` elements that allow users to switch back and forth between carousel items. They can be individually styled and also provide semantic meaning. Such elements are also compatible with assistive technology and keyboard use. 
 
-{::nomarkdown}
-<%= notes_end %>
-{:/nomarkdown}
+In the example below, JavaScript is used to generate buttons and insert them into the carousel. These particular buttons are visually displayed as arrows that overlay the carousel items.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/carousel/DOM
s/These particular buttons are visually displayed as arrows that overlay the carousel items./Visually the buttons appear as arrows overlaying the carousel items.

File: source/carousels/functionality.html.erb.md [hunk 6]
```diff
@@ -247,7 +230,9 @@ carousel.appendChild(ctrls);
 <%= code_end %>
 {:/nomarkdown}
 
-Additionally, a WAI-ARIA live region should be used to inform screen reader users which slide is now visible. Note that the current slide does not receive focus when activation the previous or next button as a user might want to skip several slides without navigating to the controls after every slide change.
+## Announcing slides
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/Announcing slides/Announce the current slide/

File: source/carousels/functionality.html.erb.md [hunk 7]
```diff
@@ -247,7 +230,9 @@ carousel.appendChild(ctrls);
...
+
+When the current item changes, a WAI-ARIA live region should be used to inform screen reader users which slide is currently visible. Note that the current slide is not activated by moving the keyboard focus there. This is done to allow users to skip several items at once.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/should be used to inform/informs/
s/Note that the current slide is not activated by moving the keyboard focus there. This is done to allow users to skip several items at once./Note that the current slide does not receive keyboard focus in this case to allow users to skip several items at once.

File: source/carousels/functionality.html.erb.md [hunk 8]
```diff
@@ -510,12 +495,12 @@ Combined, the carousel looks and functions like this:
 <%= sample_end %>
 {:/nomarkdown}
 
-## Items indicator
+## Items navigation
 {:.newex}
 
-Indicating the total number of carousel items and which one of them is currently being displayed helps users to orient themselves. Ideally each carousel item has a corresponding button that is styled acording to the state of the carousel item. Providing these buttons within a list, adds meaning and semantics to the content, such as the number and order of the carousel items.
+Display a list of buttons that indicate which carousel item is currently displayed and allow users to switch to any carousel item.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/item./item directly.

File: source/carousels/functionality.html.erb.md [hunk 9]
```diff
@@ -510,12 +495,12 @@ Combined, the carousel looks and functions like this:
...
 
-The list with buttons below is added to the carousel using JavaScript and then styled: The buttons are numbered matching the corresponding carousel items. The button corresponding to the currently displayed carousel item is highlighted both visually and using hidden text. Also the button that currently has the keyboard or mouse focus is highlighted.
+In the example below, the list with buttons is added to the carousel using JavaScript and then styled: The buttons are numbered matching the corresponding carousel items. The button for the active carousel item is highlighted both visually and using hidden text.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/and using/and by using/

File: source/carousels/functionality.html.erb.md [hunk 10]
```diff
@@ -613,9 +598,11 @@ The list with buttons below is added to the carousel using JavaScript and then s
 
 ### Focusing carousel items
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/Focusing carousel items/Focus on the currently active content

File: source/carousels/functionality.html.erb.md [hunk 11]
```diff
@@ -613,9 +598,11 @@ The list with buttons below is added to the carousel using JavaScript and then s
...
 
-In contrast to the previous/next buttons, when users select a carousel item directly through the item buttons, the focus should be set to the corresponding carousel item.
+Whe a user selects an item through the carousel navigation it should be focused by default. A keyboard or assistive technology user is sent to the item immediately.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/ it should be focused by default/, it should receive focus immediately/
s/is sent to the item immediately/is then able to interact with the content of the item right away.

File: source/carousels/functionality.html.erb.md [hunk 12]
```diff
@@ -613,9 +598,11 @@ The list with buttons below is added to the carousel using JavaScript and then s
...
+
+The `<li>` elements used for the carousel items itself are not focusable by default. However if the `tabindex` attribute with a value `-1` is added to the items, JavaScript is able to set the focus to those items.
```
Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/The `<li>` elements used for the carousel items itself are not focusable by default./The `<li>` elements used to wrap the carousel items cannot receive focus by default.

Comment (yatil) [inline source/carousels/functionality.html.erb.md]: s/ to those items//

**Comment by yatil** [comment]
**Currently “Approved Draft”:** https://www.w3.org/WAI/tutorials/carousels/functionality/
**Changes from this PR:** https://w3c.github.io/wai-tutorials/carousels/functionality/

You can comment and/or approve the changes on this page with the diff of the file(s): https://github.com/w3c/wai-tutorials/pull/318/files

**Note:** Use the blue [+] button to comment on specific lines of the page. You can then use the top-right green button to just comment or request changes. Alternatively you are also able to outright approve the changes there. You can find more information about [Pull Request Reviews in Github’s help](https://help.github.com/articles/about-pull-request-reviews/).

File: source/carousels/functionality.html.erb.md [hunk 13]
```diff
@@ -9,19 +9,14 @@ wcag_success_criteria:
   - 4.1.2
 ---
 
-Provide functionality using scripting to display the carousel items one at a time, and to allow users to browse through them.
+Carousels display the one of multiple content items at a time while hiding the others. They allow users to browse through them.
```
Comment (susanatx) [inline source/carousels/functionality.html.erb.md]: Suggestion

Carousels display multiple content items one at a time, allowing users to browse through all of them in turn.

**Pull request merged**